invokeai/backend/stable_diffusion/diffusion/custom_atttention.py

```diff
@@ -153,29 +159,34 @@ def __call__(
 
                     # Expected ip_hidden_state shape: (batch_size, num_ip_images, ip_seq_len, ip_image_embedding)
 
-                    ip_key = ipa_weights.to_k_ip(ip_hidden_states)
-                    ip_value = ipa_weights.to_v_ip(ip_hidden_states)
+                    if not self._ip_adapter_attention_weights[ipa_index].skip:
+                        if ipa_weights:
```

Comment:
We definitely should not have an `if ipa_weights` condition here (if anything it should have been an `assert ipa_weights is not None`. I'm guessing that this was just added to address a type error, because `IPAdapterAttentionWeights.ip_adapter_weights` was incorrectly marked as optional. We should be able to remove this once that is fixed.

Reply:
The reason it was marked as Optional was because there are instances where we do not pass any. `attn_procs[name] = CustomAttnProcessor2_0()` and then the classes that follow were updated to accommodate that.

Reply:
Actually resolved too.